include/boost/geometry/util/math.hpp

```diff
@@ -525,6 +525,30 @@ struct rounding_cast<Result, Source, true, false>
     }
 };
 
+template <typename T, bool IsIntegral = std::is_integral<T>::value>
+struct divide
+{
+    static inline T apply(T const& n, T const& d)
+    {
+        return n / d;
```

Comment:
what if `d==0` ?

Reply:
Then the result is as it was before so +/-Inf or NaN. But AFAIU `denominator` in segment ratio is never 0.

Reply:
Indeed, the behavior is as before.